src/main/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingImpl.java

```diff
@@ -154,39 +153,68 @@ protected SPARQLGraphPattern translateObject(final TriplePattern tp){
 		for (final Triple m : getMappings(tpQuery)) {
 			String predicate = m.asJenaTriple().getPredicate().getURI();
 			
+			/** OLD Version
 			if (predicate.equals(OWL.sameAs.getURI()) || predicate.equals(RDFS.subClassOf.getURI()) || predicate.equals(OWL.equivalentClass.getURI())) {
+				TriplePattern translation = new TriplePatternImpl(t.getSubject(), t.getPredicate(), m.asJenaTriple().getObject());
+				resultsList.add(translation);	
+			**/
+			
+			if(predicate.equals(OWL.sameAs.getURI())) {
 				TriplePattern translation = new TriplePatternImpl(t.getSubject(), t.getPredicate(), m.asJenaTriple().getObject());
 				resultsList.add(translation);
+			} else if (predicate.equals(OWL.equivalentClass.getURI())) {
 				
-			} else if (predicate.equals(OWL.unionOf.getURI())){
-				/**
-				 * TODO: How are the union represented as triples
-				 * = How to add multiple object to a triple, do I represent them with multiple triples?
-				 */
-				/** 
-				 * List<SPARQLGraphPattern> unionList = Collections.<SPARQLGraphPattern>emptyList();
-				 * for(i : m.asJenaTriple().getObject()){
-				 * 		TriplePattern translation = new TriplePatternImpl(t.getSubject(), t.getPredicate(), i);
-				 * 		unionList.add(translation)
-				 * }
-				 * SPARQLUnionPattern union = new SPARQLUnionPatternImpl(unionList);
-				 * results.add(union);
+				/*
+				 * Blank node used to help create intersections and unions
+				 * (a equals b)
+				 * (b unionof c)
+				 * (b unionof d)
 				 */
+				if (m.asJenaTriple().getObject().isBlank()) {
+					//Union or intersection
+					Node blankP = NodeFactory.createVariable("p");
```

Comment:
If you intend to create blank nodes, you should use `NodeFactory.createBlankNode("p")`

Reply:
I did not mean to create blank nodes here, but I changed the names of the nodes since I realised that they were a bit confusing.